packages/components/src/components/ScalarCombobox/ScalarComboboxOptions.vue

```diff
@@ -37,37 +38,37 @@ function getOptionId(option: Option) {
 }
 
 /** A flat list of all options */
-const options = computed<Option[]>(() =>
+const options = computed<O[]>(() =>
   isGroups(props.options)
     ? props.options.flatMap((group) => group.options)
     : props.options,
 )
 
 /** An list of all groups */
-const groups = computed<OptionGroup[]>(() =>
-  isGroups(props.options)
-    ? props.options
-    : [{ label: '', options: props.options }],
+const groups = computed<G[]>(
+  () =>
+    isGroups(props.options)
+      ? props.options // G extends OptionGroup<O>
+      : [{ label: '', options: props.options } as G], // G is OptionGroup<O>
```

Comment:
We know `G` is an  unextended `OptionGroup<O>` here because of the structure of the component props but I wasn't able to convince Typescript of that without using an `as`

Reply:
Maybe just add a code comment that clarifies this.